### DISCUSSION TITLE
Debug remotely using gdb mounted volume on docker

### DISCUSSION THREAD

**Discussion by mariamiawad** [coder/code-server#5445] [Q&A]
Hello, 

I'm trying to debug a simple helloworld.cpp program from code-server on docker,

the program is in a mounted volume

is it doable I've tried with gdbserver but it doesn't hit the breakpoint 

Thanks in advance

**Comment by jsjoeio** [comment]
Hmm... calling on @code-asher to see if he has any ideas (I haven't touched .cpp before).

**Reply by mariamiawad** [reply]
Hello @jsjoeio 

even if it is not a .cpp is it doable?
I mean even if it's python/java or any other language is it doable?

**Reply by code-asher** [reply]
It should be doable for most languages.

Unfortunately with C++ the required extensions are proprietary.  Microsoft does not allow them to be used in code-server: https://github.com/coder/code-server/discussions/5227

I am not sure if there are open-source versions of the extensions.

**Reply by lg-nineoneninethreenine** [reply]
You can download the extension VSIX file from the Microsoft proprietary marketplace and instal the extension, right?

**Reply by mariamiawad** [reply]
No need to download it from Microsoft clang is already on code-server

**Reply by code-asher** [reply]
You can download the extension manually and install into code-server but doing that is against their terms of service.
